src/ccl_halofit.c

```diff
@@ -0,0 +1,941 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+#include <gsl/gsl_integration.h>
+#include <gsl/gsl_interp.h>
+#include <gsl/gsl_spline.h>
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_roots.h>
+
+#include "ccl.h"
+
+/*
+ * We use the remapping procedure in https://arxiv.org/pdf/1601.07230.pdf
+ * to map w0-wa models onto wa == 0 models.
+ * The functions below implement this procedure.
+*/
+
+// I snuck a private function from backgrounc.c into here. :P
+void compute_chi(double a, ccl_cosmology *cosmo, double * chi, int * stat);
+
+static double zdrag_eh(ccl_parameters *params) {
```

Comment:
I wonder if we should put this in ccl_eh.c and call it from there to avoid repeating code

Reply:
I thought of this, but decided against it. Basically, there is no reason to couple the two modules together. Also any other approximation for zdrag would work fine here too. I just happened to pull the EH one, but we could have used CLASS.

Reply:
Another option would be to put it in background.c and make it visible throughout the whole library. Things like this may come in handy at some point for BAO likelihoods. We can reassess then if something similar comes up.